daprdocs/content/en/getting-started/quickstarts/pubsub-quickstart.md

```diff
@@ -0,0 +1,605 @@
+---
+
+Let's take a look at Dapr's [Publish and Subscribe (Pub/sub) building block]({{< ref pubsub >}}). In this quickstart, you will set up a publisher microservice and a subscriber microservice to demonstrate how Dapr enables a Pub/sub pattern.
+
+1. Using a publisher service, developers can repeatedly publish messages to a topic.
+1. [A Pub/sub component](https://docs.dapr.io/concepts/components-concept/#pubsub-brokers) queues or brokers those messages (using Redis Streams, RabbitMQ, Kafka, etc.).
+1. The subscriber to that topic pulls messages from the queue and processes them.
+
+Select your preferred language before proceeding with the quickstart.
+
+{{< tabs "Python" ".NET" "JavaScript" >}}
+ <!-- Python -->
+{{% codetab %}}
+
+### Pre-requisites
+
+For this example, you will need:
+
+- [Dapr CLI and initialized environment](https://docs.dapr.io/getting-started).
+- [Python 3.7+ installed](https://www.python.org/downloads/).
+
+### Step 1: Set up the environment
+
+Clone the sample we've provided.
+
+```bash
+git clone https://github.com/dapr/quickstarts.git
+```
+
+Navigate to the Pub/sub python project directory:
+
+```bash
+cd pub_sub/python
+```
+
+Install the dependencies:
+
+```bash
+pip3 install -r requirements.txt
+```
+
+Or:
+
+```bash
+python -m pip install -r requirements.txt
+```
+
+### Step 2: View the Pub/sub component
+
+When you run `dapr init`, Dapr creates a default redis `pubsub.yaml` on your local machine. Verify by opening your components directory:
+
+- On Windows, under `%UserProfile%\.dapr\components\pubsub.yaml`
+- On Linux/MacOS, under `~/.dapr/components/pubsub.yaml`
+
+For this quickstart, we've included a default Redis `pubsub.yaml` file within the cloned repository that contains the following:
+
+{{< tabs "Self-Hosted (CLI)" "Kubernetes" >}}
+
+{{% codetab %}}
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: order_pub_sub
+spec:
+  type: pubsub.redis
+  version: v1
+  metadata:
+  - name: redisHost
+    value: localhost:6379
+  - name: redisPassword
+    value: ""
+scopes:
+  - orderprocessing
+  - checkout
+```
+
+{{% /codetab %}}
+
+{{% codetab %}}
+
+Run `kubectl apply -f pubsub.yaml` to apply the following:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: order_pub_sub
+spec:
+  type: pubsub.redis
+  version: v1
+  metadata:
+  - name: redisHost
+    value: redis-master.default.svc.cluster.local:6379
+  - name: redisPassword
+    value: ""
+scopes:
+  - orderprocessing
+  - checkout
+```
+
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+### Step 3: Subscribe to topics
+
+Navigate to the directory containing your subscriber application.
+
+```bash
+cd pub_sub/python
+```
+
+With the following command, run the CheckoutService.py service alongside a Dapr sidecar.
+
+```bash
+dapr run --app-id checkout --app-port 6002 --dapr-http-port 3602 --app-protocol grpc --components-path ../components  python3 CheckoutService.py
```

Comment:
if app protocol is grpc, i'm not sure dapr-http-port is required. Instead, I would imagine dapr-grpc-port is needed